### PR TITLE
Added code and test for DeleteUser from quota server

### PR DESCRIPTION
**Pull request by poe9695**

### CODE REVIEW

File: src/main/java/edu/rosehulman/quota/Database.java [hunk 1]
```diff
@@ -27,7 +27,8 @@ public static synchronized Database getInstance() throws Exception {
   }
 
   public Optional<Partner> getPartner(String partnerId) throws Exception {
-    List<Partner> partners = getPartnerDao().query(getPartnerDao().queryBuilder().where().eq("partner_id", partnerId).prepare());
+    List<Partner> partners = getPartnerDao()
+        .query(getPartnerDao().queryBuilder().where().eq("partner_id", partnerId).prepare());
```
Comment (laeschjs) [inline src/main/java/edu/rosehulman/quota/Database.java]: Increase your line character limit for ctrl + shift + f so lines don't wrap.

File: src/main/java/edu/rosehulman/quota/Database.java [hunk 2]
```diff
@@ -63,14 +67,20 @@ public void addTier(Tier tier) throws Exception {
...
     return users.isEmpty() ? Optional.empty() : Optional.ofNullable(users.get(0));
   }
 
   public void addUser(User user) throws Exception {
     getUserDao().create(user);
   }
 
+  // TODO does this work or need to extract parts?
+  public void deleteUser(User user) throws Exception {
+    getUserDao().delete(user);
```
Comment (jackhallam) [inline src/main/java/edu/rosehulman/quota/Database.java]: This probably will not work. Do a test and see.

File: src/main/java/edu/rosehulman/quota/client/SharedServiceClient.java [hunk 3]
```diff
@@ -17,12 +17,26 @@ public static synchronized SharedServiceClient getInstance() {
   public boolean addUser(String partnerId, String productId, String userId) {
     try {
       // TODO: Put Shared Server path in config and in here
-      HttpResponse<String> response = Unirest.post("http://localhost:8084/partner/{partnerId}/product/{productId}/user/{userId}")
+      HttpResponse<String> response = Unirest
+          .post("http://localhost:8084/partner/{partnerId}/product/{productId}/user/{userId}")
```
Comment (jackhallam) [inline src/main/java/edu/rosehulman/quota/client/SharedServiceClient.java]: Line wrapping

File: src/main/java/edu/rosehulman/quota/client/SharedServiceClient.java [hunk 4]
```diff
@@ -17,12 +17,26 @@ public static synchronized SharedServiceClient getInstance() {
...
   }
+
+  public boolean deleteUser(String partnerId, String productId, String userId) {
+    try {
+      // TODO: Put Shared Server path in config and in here
+      HttpResponse<String> response = Unirest
+          .delete("http://localhost:8084/partner/{partnerId}/product/{productId}/user/{userId}")
+          .routeParam("partnerId", partnerId).routeParam("productId", productId).routeParam("userId", userId)
+          .asString();
+      return response.getStatus() == 200; // or 202???
```
Comment (jackhallam) [inline src/main/java/edu/rosehulman/quota/client/SharedServiceClient.java]: 200

File: src/main/java/edu/rosehulman/quota/controller/DeleteUserController.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    String productId = request.params(":productId");
+    String userId = request.params(":userId");
+
+    User user = new User();
+    user.setPartnerId(partnerId);
+    user.setProductId(productId);
+    user.setUserId(userId);
+
+    // delete the given user
+    Database.getInstance().deleteUser(user);
```
Comment (jackhallam) [inline src/main/java/edu/rosehulman/quota/controller/DeleteUserController.java]: If the user does not exist we want to return a 404. `deleteUser` returns a `boolean`, so check the boolean and if it is false, return a 404.

File: src/test/java/edu/rosehulman/quota/DeleteUserControllerTest.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+@PrepareForTest({ Database.class, Request.class, Response.class, SharedServiceClient.class })
+public class DeleteUserControllerTest {
+
+  @Test
+  public void testDeleteUser() throws Exception {
+    mockStatic(Database.class);
+    Database database = mock(Database.class);
+    when(Database.getInstance()).thenReturn(database);
+
+    DeleteUserController ctrller = new DeleteUserController();
```
Comment (jackhallam) [inline src/test/java/edu/rosehulman/quota/DeleteUserControllerTest.java]: `deleteUserController`

Yay auto complete

**Pull request closed without merging**